feat: Disable location sharing for C1 - WPB-9749

KaterinaWire opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9749" title="WPB-9749" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-9749</a>  [iOS] Location sharing should be disabled in C1 builds
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove the jira markers to link tickets automatically -->


### Issue
Disable location sharing for C1

**Needs releases with:**
https://github.com/wireapp/wire-ios-build-assets/pull/114

### Testing
If `LOCATION_SHARING_ENABLED` is false, the user shouldn't see the location button.

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

KaterinaWire committed:
add locationSharing security flag

KaterinaWire committed:
support LOCATION_SHARING_ENABLED flag

netbe commented on wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift:
```diff
@@ -252,9 +253,20 @@ final class ConversationInputBarViewController: UIViewController,
         return buttonsArray
     }
 
+    private var shouldExcludeLocationButton: Bool {
```
```suggestion
    /// Remove locationButton if security flag does not allow it
    private var shouldExcludeLocationButton: Bool {
```

netbe commented on wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift:
```diff
@@ -252,9 +253,20 @@ final class ConversationInputBarViewController: UIViewController,
             buttonsArray.insert(hourglassButton, at: buttonsArray.startIndex)
         }
 
+        // Remove locationButton if security flag does not allow it
```
```suggestion
```

netbe reviewed (approved):
LGTM;) don't forget to update wire-ios-build-assets too? added a small suggestion

KaterinaWire committed:
Merge branch 'release/cycle-3.112' into feat/disable-location-sharing

KaterinaWire committed:
Merge branch 'release/cycle-3.112' into feat/disable-location-sharing

KaterinaWire committed:
Update wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift

Co-authored-by: François Benaiteau <netbe@users.noreply.github.com>

KaterinaWire committed:
Update wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift

Co-authored-by: François Benaiteau <netbe@users.noreply.github.com>

KaterinaWire committed:
update wire-ios-build-assets version

KaterinaWire commented:
> LGTM;) don't forget to update wire-ios-build-assets too? added a small suggestion

I updated `wire-ios-build-assets` version to 42f724c (the last commit with merged PR)

KaterinaWire committed:
update wire-ios-build-assets to release/cycle-3.112 version

KaterinaWire committed:
Merge branch 'release/cycle-3.112' into feat/disable-location-sharing
